### PR TITLE
Draft of programming an AEA, proof read overall

### PR DESCRIPTION
**Pull request by katharine-murphy**
## Proposed changes

Including a DRAFT page of instructions to fix: https://github.com/fetchai/agents-aea/issues/298.
Complete proof read of everything from last week.

## Fixes

None.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

None.

### CODE REVIEW

File: docs/hacking-an-agent.md [hunk 1]
```diff
@@ -1,6 +1,200 @@
...
+node = LocalNode()
+```
+
+Initialise a `MailBox` object with a public key and grab the private key and add it to a wallet.
+
+``` python
+public_key_1 = "mailbox1"
+mailbox1 = MailBox(OEFLocalConnection(public_key_1, node))
+private_key_pem_path = os.path.join(CUR_PATH, "data", "priv.pem")
+wallet = Wallet({'default': private_key_pem_path})
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: The wallet creation should be done first.

Then they can get the public key like so:
`public_key = wallet.public_keys['default']`

Then they can set up a mailbox like so:
`mailbox = MailBox(stub_connection)`

File: docs/hacking-an-agent.md [hunk 2]
```diff
@@ -1,6 +1,200 @@
...
+from aea.registries.base import Resources
+from aea.skills.base import Skill
+from .conftest import CUR_PATH, DummyConnection
+```
+
+
+
+## Initialise the agent
+
+Create a node.
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: Let's use the stub connection instead:
`stub_connection = StubConnection(input_file_path='input.txt', output_file_path='output.txt')

File: docs/hacking-an-agent.md [hunk 3]
```diff
@@ -1,6 +1,200 @@
...
+public_key_1 = "mailbox1"
+mailbox1 = MailBox(OEFLocalConnection(public_key_1, node))
+private_key_pem_path = os.path.join(CUR_PATH, "data", "priv.pem")
+wallet = Wallet({'default': private_key_pem_path})
+```
+
+Using a variable for accessing running ledgers, initialise the agent. 
+
+``` python
+ledger_apis = LedgerApis({})
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: Separate this out from the aea initialisation. Just say that we do not add any ledger apis by feeding it a dict `{}`

File: docs/hacking-an-agent.md [hunk 4]
```diff
@@ -1,6 +1,200 @@
...
+my_AEA = AEA("Agent0", mailbox1, wallet, ledger_apis, resources=Resources(str(Path(CUR_PATH, "aea"))))
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: let's give the agent a better name :)

also, let's instantiate the resources separately:
`resources = Resources(str(Path(CUR_PATH, "aea")))`

File: docs/hacking-an-agent.md [hunk 5]
```diff
@@ -1,6 +1,200 @@
...
+```
+
+
+## Run the agent
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: Remove lines 70 - 78

File: docs/hacking-an-agent.md [hunk 6]
```diff
@@ -1,6 +1,200 @@
...
+
+``` python
+behaviour = agent.resources.behaviour_registry.fetch("dummy")
+assert behaviour[0].nb_act_called > 0, "Act() wasn't called"
+```
+
+Finalise the agent thread.
+
+``` python
+finally:
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: remove finally line.

File: docs/hacking-an-agent.md [hunk 7]
```diff
@@ -1,6 +1,200 @@
...
+	agent.stop()
+	t.join()
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: add t = None

File: docs/hacking-an-agent.md (hunk 1 above)
Comment (DavidMinarsch) [reply docs/hacking-an-agent.md]: True, I would still swap it around as in all but this connection case the public key is needed and derived from the wallet.

File: docs/hacking-an-agent.md [hunk 8]
```diff
@@ -43,39 +43,41 @@ from .conftest import CUR_PATH, DummyConnection
...
+Create a wallet with a private key.
 ``` python
-public_key_1 = "mailbox1"
-mailbox1 = MailBox(OEFLocalConnection(public_key_1, node))
 private_key_pem_path = os.path.join(CUR_PATH, "data", "priv.pem")
 wallet = Wallet({'default': private_key_pem_path})
 ```
 
-Using a variable for accessing running ledgers, initialise the agent. 
+Get the public key from the wallet. You would use this to talk to an OEF or Fetch.ai ledger node.
```
Comment (DavidMinarsch) [inline docs/hacking-an-agent.md]: I would comment this out. We might use it another time but here it does not add value.

Comment (katharine-murphy) [reply docs/hacking-an-agent.md]: I put the pk comments in a Note. Awaiting your approval.

**Pull request merged**